modules/criteortusIdSystem.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * This module adds Criteo Real Time User Sync to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/criteortusIdSystem
+ * @requires module:modules/userId
+ */
+
+import * as utils from '../src/utils'
+import { ajax } from '../src/ajax';
+import { attachIdSystem } from '../modules/userId';
+
+const key = '__pbjs_criteo_rtus';
+
+/** @type {Submodule} */
+export const criteortusIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: 'criteortus',
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @returns {{criteortus:Object}}
+   */
+  decode() {
+    let uid = utils.getCookie(key);
+    try {
+      uid = JSON.parse(uid);
+      return { 'criteortus': uid };
+    } catch (error) {
+      utils.logError('Error in parsing criteo rtus data', error);
+    }
+  },
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleParams} [configParams]
+   * @returns {function(callback:function)}
+   */
+  getId(configParams) {
+    if (!configParams || !utils.isPlainObject(configParams.clientIdentifier)) {
+      utils.logError('User ID - Criteo rtus requires client identifier to be defined');
+      return;
+    }
+
+    let uid = utils.getCookie(key);
+    if (uid) {
+      return uid;
+    } else {
+      let userIds = {};
+      return function(callback) {
+        let bidders = Object.keys(configParams.clientIdentifier);
+
+        function afterAllResponses() {
+          // criteo rtus user id expires in 1 hour
+          const expiresStr = (new Date(Date.now() + (60 * 60 * 1000))).toUTCString();
+          utils.setCookie(key, JSON.stringify(userIds), expiresStr);
+          callback(userIds);
+        }
+
+        const onResponse = utils.delayExecution(afterAllResponses, bidders.length);
+
+        bidders.forEach((bidder) => {
+          let url = `https://gum.criteo.com/sync?c=${configParams.clientIdentifier[bidder]}&r=3`;
```

Comment:
this isn't going to scale out so well. Is that the only way Criteo supports (1 ID at a time?)

Reply:
Yes, each bidder will have it's own client identifier. As per information we have now, this seems only option for now.  There will be changes once we know more about this.